### PR TITLE
Debugger panel improvements

### PR DESCRIPTION
**Pull request by jackTabsCode**
This PR includes improvements to the debugger panel, in specific the systems section:
- Bars under the systems indicating their duration relative to the longest system. A full bar means this system is taking the most time (but doesn't necessarily mean it's not performant).
- Index to the left of the system name for improved visual grepping
- Panel slightly wider to accodomate the index element

Also a minor font change in an unrelated file is included since I forgot it in my last style PR.

RobloxStudioBeta_mcQF7dZ6Zq

### CODE REVIEW

File: lib/debugger/widgets/selectionList.lua [hunk 1]
```diff
@@ -76,15 +107,17 @@ return function(Plasma)
...
+			refs.button.container.TextLabel.Text = text
+			refs.button.container.Icon.Text = icon or ""
+			refs.button.container.Icon.Visible = not not icon
 		end, text, icon)
 
-		refs.sideText.Visible = not not sideText
-		refs.sideText.Text = sideText or ""
-		refs.sideText.TextColor3 = if selected then style.textColor else style.mutedTextColor
-		refs.button.TextLabel.TextTruncate = sideText and Enum.TextTruncate.AtEnd or Enum.TextTruncate.None
+		refs.button.container.sideText.Visible = not not sideText
```
Comment (benbrimeyer) [inline lib/debugger/widgets/selectionList.lua:115]:
```suggestion
		refs.button.container.sideText.Visible = sideText ~= nil
```
the `not not`'s are hard to parse what the actual intent is

File: lib/debugger/widgets/selectionList.lua [hunk 2]
```diff
@@ -76,15 +107,17 @@ return function(Plasma)
 		end)
 
 		Plasma.useEffect(function()
-			refs.button.TextLabel.Text = text
-			refs.button.Icon.Text = icon or ""
-			refs.button.Icon.Visible = not not icon
+			refs.button.container.TextLabel.Text = text
+			refs.button.container.Icon.Text = icon or ""
+			refs.button.container.Icon.Visible = not not icon
```
Comment (benbrimeyer) [inline lib/debugger/widgets/selectionList.lua:112]:
```suggestion
			refs.button.container.Icon.Visible = icon ~= nil
```

File: lib/debugger/widgets/selectionList.lua [hunk 3]
```diff
@@ -76,15 +107,17 @@ return function(Plasma)
...
 		end, text, icon)
 
-		refs.sideText.Visible = not not sideText
-		refs.sideText.Text = sideText or ""
-		refs.sideText.TextColor3 = if selected then style.textColor else style.mutedTextColor
-		refs.button.TextLabel.TextTruncate = sideText and Enum.TextTruncate.AtEnd or Enum.TextTruncate.None
+		refs.button.container.sideText.Visible = not not sideText
+		refs.button.container.sideText.Text = sideText or ""
```
Comment (benbrimeyer) [inline lib/debugger/widgets/selectionList.lua:116]:
```suggestion
		refs.button.container.sideText.Text = if sideText ~= nil then sideText else ""
```
more of a nitpick, if sideText is ever a false, we won't render anything.

File: lib/debugger/ui.lua [hunk 4]
```diff
@@ -160,6 +169,8 @@ local function ui(debugger, loop)
 						selected = debugger.debugSystem == system,
 						system = system,
 						icon = icon,
+						barWidth = duration / longestDuration,
```
Comment (benbrimeyer) [inline lib/debugger/ui.lua:172]: off hand, do we need to worry about anything if longestDuration is 0?

File: lib/debugger/widgets/selectionList.lua (hunk 1 above)
Comment (jackTabsCode) [reply lib/debugger/widgets/selectionList.lua:115]: Sure. These expressions were brought over from old code.

File: lib/debugger/ui.lua (hunk 4 above)
Comment (jackTabsCode) [reply lib/debugger/ui.lua:172]: No, because longestDuration will get set to a non-zero number if there's a system, and this code is in a loop over systems.

Comment (benbrimeyer) [reply lib/debugger/ui.lua:172]: too clever for our own good

File: lib/debugger/widgets/selectionList.lua (hunk 1 above)
Comment (benbrimeyer) [reply lib/debugger/widgets/selectionList.lua]: Too clever for our own good

File: lib/debugger/ui.lua (hunk 4 above)
Comment (benbrimeyer) [reply lib/debugger/ui.lua:172]: its not very clear there's a implicit dependency on that fact with the given control flow, but I do believe its unlikely to happen

File: lib/debugger/widgets/selectionList.lua [hunk 5]
```diff
@@ -125,7 +158,9 @@ return function(Plasma)
 		local selected
 
 		for _, item in items do
-			if Item(item.text, item.selected, item.icon, item.sideText, options.width):clicked() then
+			if
+				Item(item.text, item.selected, item.icon, item.sideText, options.width, item.barWidth, item.index):clicked()
```
Comment (benbrimeyer) [inline lib/debugger/widgets/selectionList.lua:162]: should we remove options.width here since it appears unused?

Comment (jackTabsCode) [reply lib/debugger/widgets/selectionList.lua:162]: We could be here all day cleaning up the debugger code, but I would like to retain the scope of the PR

Comment (benbrimeyer) [reply lib/debugger/widgets/selectionList.lua:162]: It feels within scope since you are adding new arguments

Comment (benbrimeyer) [reply lib/debugger/widgets/selectionList.lua:162]: My previous question was an honest one not being familiar with plasma, but the `_` argument up top clued me in. If it is true we dont use this argument, I can make the commit for you and clean as we go

Comment (jackTabsCode) [reply lib/debugger/widgets/selectionList.lua:162]: It is used by `worldInspect.lua`, whether it is useful or not

Comment (benbrimeyer) [reply lib/debugger/widgets/selectionList.lua:162]: https://github.com/matter-ecs/matter/pull/35

**Review by LastTalon — APPROVED** [review]
LGTM
